src/darker/__main__.py

```diff
@@ -136,9 +136,13 @@ def modify_file(path: Path, new_content: TextDocument) -> None:
     path.write_bytes(new_content.encoded_string)
 
 
-def print_diff(path: Path, old: TextDocument, new: TextDocument) -> None:
+def print_diff(
+    path: Path, old: TextDocument, new: TextDocument, root: Optional[Path] = None
```

Comment:
For keyword arguments with a default value of `None`, actually `Optional[...]` can be omitted and it is assumed by Mypy due to  the default value.

That said, I'm not sure if it's better to write `root: Path = None` or `root: Optional[Path] = None`. What do you think?

```suggestion
def print_diff(
    path: Path, old: TextDocument, new: TextDocument, root: Path = None
```